packages/js/api/src/framework/__tests__/payload-transformer.spec.ts

```diff
@@ -0,0 +1,383 @@
+						minimum_amount: '',
+						maximum_amount: '',
+						email_restrictions: [],
+						virtual: false,
+						meta_data: []
+					},
+					display_key: 'coupon_data',
+					display_value: {
+						id: 171,
+						code: 'save5',
+						amount: '5',
+						date_created: {
+							date: '2021-05-19 04:28:31.000000',
+							timezone_type: 3,
+							timezone: 'Pacific/Auckland'
+						},
+						date_modified: {
+							date: '2021-05-19 04:28:31.000000',
+							timezone_type: 3,
+							timezone: 'Pacific/Auckland'
+						},
+						date_expires: null,
+						discount_type: 'fixed_cart',
+						description: '',
+						usage_count: 3,
+						individual_use: false,
+						product_ids: [],
+						excluded_product_ids: [],
+						usage_limit: 0,
+						usage_limit_per_user: 0,
+						limit_usage_to_x_items: null,
+						free_shipping: false,
+						product_categories: [],
+						excluded_product_categories: [],
+						exclude_sale_items: false,
+						minimum_amount: '',
+						maximum_amount: '',
+						email_restrictions: [],
+						virtual: false,
+						meta_data: []
+					}
+				}
+			]
+		}
+	],
+	refunds: [
+		{
+			id: 123,
+			reason: 'Foo Reason',
+			total: '5.00'
+		}
+	],
+	date_created_gmt: '2021-11-30T16:43:38',
+	date_modified_gmt: '2021-11-30T18:31:05',
+	date_completed_gmt: null,
+	date_paid_gmt: null,
+	currency_symbol: '$',
+	_links: {
+		self: [
+			{
+				href: 'http://local.wordpress.test/wp-json/wc/v3/orders/1218'
+			}
+		],
+		collection: [
+			{
+				href: 'http://local.wordpress.test/wp-json/wc/v3/orders'
+			}
+		]
+	}
+});
+
+describe( 'OrderTransformer', () => {
+	it( 'should transform an order JSON', () => {
+		const order = createOrderTransformer().toModel( Order, JSON.parse(jsonPayloadFull) );
+		const billing = createBillingAddressTransformer().toModel(BillingOrderAddress, JSON.parse(jsonPayloadFull).billing);
+		const shipping = createShippingAddressTransformer().toModel(ShippingOrderAddress, JSON.parse(jsonPayloadFull).shipping);
+
+		// Order
+		expect( order ).toBeInstanceOf( Order );
+		expect( order.id ).toStrictEqual(1218);
+		expect( order.parentId ).toStrictEqual(0);
+		expect( order.status ).toStrictEqual('pending');
+		expect( order.currency ).toStrictEqual('USD');
+		expect( order.version ).toStrictEqual('6.1.0');
+		expect( order.pricesIncludeTax ).toStrictEqual(false);
+		//expect( model.dateCreated ).toStrictEqual('2021-12-01T05:43:38');
+		//expect( model.dateModified ).toStrictEqual('2021-12-01T07:31:05');
+		expect( order.discountTotal ).toStrictEqual('5.00');
+		expect( order.discountTax ).toStrictEqual('0.60');
+		expect( order.shippingTotal ).toStrictEqual('0.00');
+		expect( order.shippingTax ).toStrictEqual('0.00');
+		expect( order.cartTax ).toStrictEqual('6.00');
+		expect( order.total ).toStrictEqual('56.00');
+		expect( order.totalTax ).toStrictEqual('6.00');
+		expect( order.customerId ).toStrictEqual(0);
+		//expect( model.orderKey ).toStrictEqual('wc_order_GFHWVmAiamh0B');
```

Comment:
Based on our conversations, you mentioned that the commented test steps weren't working,  you should be able to to work by adding the appropriate key and property type transformations. 

```suggestion
		expect( order.orderKey ).toStrictEqual('wc_order_GFHWVmAiamh0B');
```

With the suggested change above, you could add `orderKey: 'order_key'` to the [key transformer](https://github.com/woocommerce/woocommerce/blob/8e98a7c1e36d5c10928a3a61c9b9237a4c2edd45/packages/js/api/src/repositories/rest/orders/transformer.ts#L56) and `orderKey: PropertyType.String` to the [property type transformer](https://github.com/woocommerce/woocommerce/blob/8e98a7c1e36d5c10928a3a61c9b9237a4c2edd45/packages/js/api/src/repositories/rest/orders/transformer.ts#L37) and the test should pass.

Reply:
Thanks, I have left those commented out precisely to facilitate doing this.